### PR TITLE
AI_FLAG_RISKY Improvements

### PR DESCRIPTION
**Pull request by Pawkkie**
## Description
#4615 added the ability to choose what damage rolls the AI uses when scoring its moves, which introduced the concept of having the pre-existing AI_FLAG_RISKY assume it gets high rolls and score accordingly. This prompted discussing a minor overhaul to AI_FLAG_RISKY, which is otherwise just a simple list of effects that the AI will score more highly. 

After that initial discussion, we decided that a fun risky AI would:

- **Prioritize maximizing damage from moves at the cost of accuracy** (I'm interpreting this as "When comparing damaging moves, the AI only considers accuracy if expected damage is the same (ie. multiple neutral attacks with the same power), even if hits to KO are the same (if the player switches the AI still wants to do the biggest chunk of damage it can)")
- **Prioritize moves with low chance strong effects like Ancient Power** suggestions for this were crowd-sourced)
- **Use high rolls to evaluate damage calculations** (already implemented in #4615)
- **Switch offensively mid battle rather than defensively** (defensive mid-battle is the default behaviour universally, see GetBestMonIntegrated for details)
- **Enjoy using explosion moves** (this seems important to keep for some users)

The list of effects that should be prioritized is likely going to expand and change over time as we have more ideas, but as a starting point that list includes the following:

**Strongly Prioritize (+3)**
- EFFECT_COUNTER if player mon is physical attacker (if species' base Atk is 10+ higher than SpAtk)
- EFFECT_MIRROR_COAT if player mon is special attacker (if species' base SpAtk is 10+ higher than Atk)
- EFFECT_EXPLOSION

**Moderately Prioritize (+2)**
- EFFECT_REVENGE if player mon's species has base Speed 10+ higher than AI mon's
- EFFECT_BELLY_DRUM (at 90+% HP)
- EFFECT_MAX_HP_50_RECOIL (Steel Beam)
- EFFECT_MIND_BLOWN
- EFFECT_SWAGGER
- EFFECT_FLATTER
- EFFECT_ATTRACT
- EFFECT_OHKO
- MOVE_EFFECT_ALL_STATS_UP (Ancient Power etc.)
- High critical hit ratio

This list also included considerations for Destiny Bond initially, but it seems well covered elsewhere in the scoring AI as is in my opinion.

Lastly, I've written tests for the more complex behaviours, including Mirror Coat, Counter, Revenge, mid-battle switches, and prioritizing damage at the expense of accuracy.

## **People who collaborated with me in this PR**
Feedback from @AlexOn1ine and @iriv24 in Discord

## **Discord contact info**
@pawkkie

### CODE REVIEW

**Comment by Pawkkie** [comment]
I have no idea why the check has failed by adding `case EFFECT_MIND_BLOWN:`, it passes all tests and compiles without errors on my machine.

File: src/battle_ai_main.c [hunk 1]
```diff
@@ -3217,6 +3228,59 @@ static s32 AI_CompareDamagingMoves(u32 battlerAtk, u32 battlerDef, u32 currId)
         }
     }
 
+    // Compare damage rather than hits to KO for AI_FLAG_RISKY   
+    if ((AI_THINKING_STRUCT->aiFlags[battlerAtk] & AI_FLAG_RISKY) && bestDamage == damage[currId])
```
Comment (AlexOn1ine) [inline src/battle_ai_main.c:3232]: How likely is it that more then one move will have the same damage while being the best? I wonder if it makes sense to just increase score for best damage and ignore further calcs?

**Review by AlexOn1ine — COMMENTED** [review]
One thing but looks good otherwise. I will try to investigate why Mind Blown causes the test to fail /Edit ah, it seems like CI broke again.

File: src/battle_ai_main.c (hunk 1 above)
Comment (Pawkkie) [reply src/battle_ai_main.c:3232]: I think it'd be super unlikely and I'm open to this approach for sure, I just figured I'd leave in all the niche-case code from the main approach for the sake of accuracy.

D'you think having worse niche-case behaviour is worth the cleaner code? The niche cases are gonna be pretty few and far between.

Comment (AlexOn1ine) [reply src/battle_ai_main.c:3232]: For AI I prefer cleaner code in favor of addressing very niche cases because the AI code is already huge.

**Review by AlexOn1ine — APPROVED** [review]

**Pull request merged**